cypress/page/products-list.page.ts

```diff
@@ -1,18 +1,25 @@
 class ProductListPage {
     private addToCart: string;
     private ProceedToCheckout: string
+    private productList: string
 
     constructor() {
-        this.addToCart = "#center_column a.button.ajax_add_to_cart_button.btn.btn-default";
+        this.addToCart = "a.button.ajax_add_to_cart_button.btn.btn-default";
         this.ProceedToCheckout = "[style*=\"display: block;\"] .button-container > a"
+        this.productList = "#center_column ul.product_list li"
     }
-    public goAddToCart(): void {
+    public goAddToCart(name: string): void {
+        var product = this.findProductByName(name)
         cy.get(this.addToCart).click()
     }
 
     public GoToCheckout(): void {
         cy.get(this.ProceedToCheckout).click()
     }
+
+    private findProductByName(name: string): Chainable<JQuery<HTMLElement>> {
+        return cy.get(this.productList).filter(":contains('" + name + "')")
```

Comment:
To avoid concatenate strings with the `+`.  You can use the backtick \` instead of  " and call the input param between `${}`. Example:
`:contains(${productName})`

Reply:
yeah this is called string interpolation

Reply:
Done